Cabal/Distribution/Pretty.hs

```diff
@@ -60,11 +60,12 @@ flatStyle = PP.Style { PP.mode = PP.LeftMode
 type Separator = [PP.Doc] -> PP.Doc
 
 showFilePath :: FilePath -> PP.Doc
-showFilePath "" = mempty
-showFilePath x  = showToken x
+showFilePath = showToken
 
 showToken :: String -> PP.Doc
 showToken str
+    | "--" `isPrefixOf` str                 = PP.text (show str)
```

Comment:
Would be nice with a comment here.

Reply:
I'll add a comment, but `--foo` looks like a comment.